src/preset_cli/api/clients/superset.py

```diff
@@ -564,7 +615,7 @@ def create_dataset(self, **kwargs: Any) -> Any:
 
         payload = response.json()
 
-        return payload["data"]
+        return payload["data"] if "data" in payload else {"id": payload["table_id"]}
```

Comment:
```suggestion
        # Superset 1.4 returns ``{"table_id": dataset_id}`` rather than the dataset payload
        return payload["data"] if "data" in payload else {"id": payload["table_id"]}
```